utils.js

```diff
@@ -34,9 +34,9 @@ module.exports = {
     hashToken: async function (token) {
 
         const saltRounds = 10;
-        const salt = await bcrypt.genSalt(saltRounds);
+        const salt = "$2b$10$tcRAZ/327/jH5r9xKslwhe";
```

Comment:
A new salt was being generated each time, causing hashes of same strings to be different each time. Equating it to 1 hash makes it easier to query the database using the token.

Reply:
This is the purpose of salt – i.e., to prevent [dictionary attacks (rainbow table attacks)](https://en.m.wikipedia.org/wiki/Dictionary_attack) by ensuring a hash of a string is not predictable. The solution isn't to defeat the security of bcrypt; rather, we'll need to come up with a way to identify a module independently of it's token (password). What do we store as the principal? Perhaps the module should generate two UUIDs: one as its ID and the other as its token. Then we could search by id (principal) and validate with token as a password.

Reply:
Thats a really robust idea! I'll start working on it. :)

Reply:
I updated both the PRs. :)